pymedphys/cli/__init__.py

```diff
@@ -63,22 +65,63 @@ def define_parser():
     return parser
 
 
-def pymedphys_cli():
-    parser = define_parser()
+def get_logging_config():
+    try:
+        config = get_config()
+    except FileNotFoundError:
+        return {}
+
+    try:
+        cli_config = config["cli"]
+        logging_config = cli_config["logging"]
+    except KeyError:
+        return {}
+
+    return logging_config
 
-    args, remaining = parser.parse_known_args()
 
-    loglevel = logging.WARNING
-    logformat = "%(asctime)s %(levelname)-8s %(message)s"
+def run_logging_basic_config(args):
+    logging_config = get_logging_config()
 
+    if "level" not in logging_config:
+        logging_config["level"] = logging.WARNING
+    else:
+        try:
+            logging_config["level"] = getattr(logging, logging_config["level"].upper())
+        except AttributeError:
+            pass
+
+    # Allow command line options to override the config.toml options
     if args.logging_verbose:
-        loglevel = logging.INFO
+        logging_config["level"] = logging.INFO
 
+    # Have debug after info so that if both --verbose and --debug are
+    # passed to the CLI debug will be used.
     if args.logging_debug:
-        loglevel = logging.DEBUG
-        logformat += "\n    %(pathname)s#%(lineno)d"
+        logging_config["level"] = logging.DEBUG
+
+    if "format" not in logging_config:
+        logging_config["format"] = "%(asctime)s %(levelname)-8s %(message)s"
 
-    logging.basicConfig(format=logformat, level=loglevel, datefmt="%Y-%m-%d %H:%M:%S")
+        if logging_config["level"] <= logging.DEBUG:
+            logging_config["format"] += "\n    %(pathname)s#%(lineno)d"
+
+    if "datefmt" not in logging_config:
+        logging_config["datefmt"] = "%Y-%m-%d %H:%M:%S"
+
+    logging.basicConfig(**logging_config)
```

Comment:
If you intend to call this method more than once during the lifetime of the cli "instance", I think you want to pass force=True.  I'm not 100% sure about that, but that's my read of
https://docs.python.org/3.8/library/logging.html#logging.basicConfig
Given that it includes "This function does nothing if the root logger already has handlers configured, unless the keyword argument force is set to True."

Given that this method will only be called once at the moment, that's not a defect.
If this method will only ever be called once, it's not a defect.
Otherwise it's a latent defect.

Reply:
Yup good point, it's not currently a defect, but if I forget to change it later, it would be a defect then. Changed :). Cheers :).